packages/venia-ui/lib/components/ProductFullDetail/productFullDetail.js

```diff
@@ -127,7 +127,7 @@ const ProductFullDetail = props => {
             // We must create a new Map here so that React knows that the value
             // of optionSelections has changed.
             const newOptionSelections = new Map([...optionSelections]);
-            newOptionSelections.set(optionId, Array.from(selection).pop());
+            newOptionSelections.set(optionId, selection);
```

Comment:
This feels better. I don't know why we were popping the return value but it felt too coupled to whatever was invoking the callback.